lib/fluent/plugin/elasticsearch_index_template.rb

```diff
@@ -35,9 +35,10 @@ def template_exists?(name, host = nil)
   def retry_operate(max_retries, fail_on_retry_exceed = true)
     return unless block_given?
     retries = 0
+    transport_errors = Elasticsearch::Transport::Transport::Errors.constants().map{ |c| Elasticsearch::Transport::Transport::Errors.const_get c }
```

Comment:
I think that all errors for transporting is awful handling.
For example, `BadRequest`, or `MethodNotAllowed`, or `UnsupportedMediaType` requests shouldn't repeat.

Reply:
@cosmo0920 
Let me explain the issue we are having. We are exposing a fluentd as a service, users can setup their own output destination to receive logs from our backend. This is working inside Kubernetes with auto scaling based on CPU. The issue is that if they do not set it correctly this crash and repeat every couple of seconds is hitting 100% CPU usage and the HorizontalPodAutoscaler starts scaling the invalid services. The error returned may as well be BadRequest from the destination they have set up.

We either need a hard crash in these cases so that the pod can simple go into a crashloop or this retry, so that their CPU usage is not 100%.

Reply:
I looked into Fluentd Output buffer code and saw that the issue comes when an error is raised in the `configure` or `start` phase and the buffer retry logic is only happening with errors during the `write(chunk)` phase. Maybe we should fork Fluentd and apply a hard crash on errors in the first two phases.

The Splunk plugin amazed me by having intentionally consuming a chunk when any of the 4xx http responses is returned: https://github.com/splunk/fluent-plugin-splunk-hec/blob/develop/lib/fluent/plugin/out_splunk_hec.rb#L308

We are using it with Kafka and consuming a chunk will be a data loss for the specific user since their consumer offset is getting committed.

Reply:
I see. Pokemon exception for `transport_errors` are reasonable for me in this case.
Thank you for the explanation.